edivorce/apps/core/migrations/0023_auto_20201006_1314.py

```diff
@@ -0,0 +1,41 @@
+# Generated by Django 2.2.15 on 2020-10-06 20:14
+
+from django.db import migrations
+
+def migrate_name_forward(apps, schema_editor):
+    UserResponse = apps.get_model('core', 'UserResponse')
+    name_you_responses = UserResponse.objects.filter(question_id='name_you')
+    print(f"Converting {name_you_responses.count()} name_you responses")
+    for response in name_you_responses:
+        response.question_id = 'last_name_you'
+        response.save()
+    name_spouse_responses = UserResponse.objects.filter(question_id='name_spouse')
+    print(f"Converting {name_spouse_responses.count()} name_spouse responses")    
+    for response in name_spouse_responses:
+        response.question_id = 'last_name_spouse'
+        response.save()
+
+
+def migrate_name_backwards(apps, schema_editor):
+    UserResponse = apps.get_model('core', 'UserResponse')
+    last_name_you_responses = UserResponse.objects.filter(question_id='last_name_you')
+    print(f"Converting {last_name_you_responses.count()} last_name_you responses")
+    for response in last_name_you_responses:
+        response.question_id = 'name_you'
```

Comment:
I don't think it's really necessary to join the fields.

given_name_1, given_name_2, given_name_3 will not exist immediately after the forward migration

They won't be created until a user fills out the form.  

If they do exist then the data won't be lost, it will only be orphanned.

Reply:
Yeah, I guess that's true. I generally think of the reverse function as being a full reversal of the original migration, so would also delete the new fields if they exist. But in reality, this functionality is almost never used. So I think it's fine as is.